src/access_nri_intake/source/builders.py

```diff
@@ -751,17 +751,30 @@ def __init__(self, path, **kwargs):
     @classmethod
     def parser(cls, file) -> dict:
         try:
-            realm = "ocean"
-
             nc_info = cls.parse_ncfile(file, time_dim="time")
             ncinfo_dict = nc_info.to_dict()
 
-            ncinfo_dict["realm"] = realm
+            ncinfo_dict = cls.post_proc_ncinfo(ncinfo_dict)
 
             return ncinfo_dict
         except Exception:
             return {INVALID_ASSET: file, TRACEBACK: traceback.format_exc()}
 
+    @staticmethod
+    def post_proc_ncinfo(ncinfo_dict: dict) -> dict:
+        """
+        Post process the ncinfo_dict to add anything that looks like `KDS` to the file id
+        """
+        file_id = ncinfo_dict["file_id"]
+        pathparts = Path(ncinfo_dict["path"]).parts
+        for part in pathparts:
+            if "kds" in part.lower():
+                file_id = f"{file_id}_{part}"
```

Comment:
Should this `break` after reaching the `if` statement? Otherwise you might get multiple `_part` added to the file_id... unless that's the intended behaviour.